webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -8,12 +8,17 @@ import { notificationsDrawer } from '../../../common/sessionStorage';
 import API from '../../../API';
 
 export const getNotifications = url => dispatch => {
-  API.get(url).then(response => dispatch({
-    type: NOTIFICATIONS_GET_NOTIFICATIONS,
-    payload: {
-      notifications: response.notifications
-    }
-  }));
+  if (
+    document.visibilityState === 'visible' ||
+    document.visibilityState === 'prerender'
+  ) {
+    API.get(url).then(response => dispatch({
```

Comment:
also, does this make sure the API calls happens only after the last completed? e.g. it will wait the TIMER interval from when the last successful call.